src/App/DataSource/Twitter/Twitter.php

```diff
@@ -187,13 +187,20 @@ public function fetch($limit = false)
                 ) {
                     continue;
                 }
+                $user_id = $user['id_str'];
                 // @todo Check for similar messages in the database before saving
                 $messages[] = [
                     'type' => MessageType::TWITTER,
                     'contact_type' => Contact::TWITTER,
-                    'from' => $user['id_str'],
+                    'from' => $user_id,
                     'to' => null,
-                    'message' => 'https://twitter.com/statuses/' . $id,
+                    /** 
+                     * Best compromise I could find was just make the proper urls with user_id rather than only 
+                     * tweet id (for which there is an unofficial formula)... since there doesn't seem to be a way to grab the URL from the 
+                     * API itself in the v1.1 search endpoint. 
+                     * Fun fact: if the user id is wrong, twitter still takes you to the correct Tweet... they just use the tweet id
+                    **/
+                    'message' => "https://twitter.com/$user_id/status/$id",
```

Comment:
couldn't find a way to get twitter urls... so I'm using this more obvious user facing URL instead of /statuses... which I suppose gives me more peace of mind... but still, would prefer if twitter actually gave us the url to use, you know?

Reply:
I'd so much love that too!